concrete/src/Attribute/Key/SearchIndexer/StandardSearchIndexer.php

```diff
@@ -32,6 +32,71 @@ protected function getIndexEntryColumn(Key $key, $subKey = false)
     }
 
     /**
+     * For certain fields (eg TEXT) Doctrine uses the length of the longest column to determine what field type to use.
+     * For search indexing even if we may not currently have something long in a column,
+     * we need the longest possible column so that we don't truncate any data.
+     *
+     * @param array $options
+     *
+     * @return array
+     */
+    private function setTypeLength($options)
+    {
+        // If we have explicitly set a length, use it
+        if ($options['length']) {
+            return $options;
+        }
+        if ($options['type']->getName() == 'text') {
+            $options['length'] = 4294967295; // This forces Doctrine to use `LONGTEXT` instead of `TINYTEXT`
+        }
+
+        return $options;
+    }
+
+    /**
+     * Refresh the Search Index columns (if there are schema changes for example).
+     *
+     * @param CategoryInterface $category
+     * @param AttributeKeyInterface $key
+     */
+    public function refreshSearchIndexKeyColumns(CategoryInterface $category, AttributeKeyInterface $key)
+    {
+        $controller = $key->getController();
+        $definition = $controller->getSearchIndexFieldDefinition();
+        $sm = $this->connection->getSchemaManager();
+        $fromTable = $sm->listTableDetails($category->getIndexedSearchTable());
+        $toTable = $sm->listTableDetails($category->getIndexedSearchTable());
+
+        if (isset($definition['type'])) {
+            $options = [
+                'type' => Type::getType($definition['type']),
+            ];
+            $options = array_merge($options, $definition['options']);
+            $options = $this->setTypeLength($options);
+            $toTable->changeColumn('ak_' . $key->getAttributeKeyHandle(), $options);
+        } else {
+            foreach ($definition as $name => $column) {
+                $options = [
+                    'type' => Type::getType($column['type']),
+                ];
+                $options = array_merge($options, $column['options']);
+                $options = $this->setTypeLength($options);
+                $toTable->changeColumn('ak_' . $key->getAttributeKeyHandle() . '_' . $name, $options);
+            }
+        }
+        $comparator = new \Doctrine\DBAL\Schema\Comparator();
```

Comment:
Makes sense, I believe that change to be outside the scope of this pull request, as there is another method in this file specifically that use this exact same call (and has some other stuff that should be abstracted out in the same way)